### PR TITLE
Create HR version of Job Review page

### LINKED ISSUE #2303: Task - HR - View of Job Poster Builder review page
# Description
Create a view of the Job Poster Builder review page for HR.

### The following should be different from the Manager View
- (if easy to do) Different steps should not be shown at the top of page. This is not critical though.
- HR commenting should be available on each section (instead of manager commenting)
- The only button that should be shown on the bottom is the "Return to Summary"
- The "Edit Poster" for managers (or "Review Poster" as it should probably be called for HR advisers) button on the Summary page should link here.

### Tests
- [x] This task requires a unit test.

### PR DESCRIPTION
**Pull request by tristan-orourke**
Resolves #2303 

### Notes:
- Made some changes to the JPB review page, to make parts of it easier to reuse. So during review, it would be worthwhile testing the JPB review page as well.

### CODE REVIEW

File: resources/assets/js/components/HRPortal/JobReviewHrPage.tsx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+const container = document.getElementById("job-review-hr");
+if (container !== null) {
+  if ("jobId" in container.dataset) {
+    const jobId = Number(container.dataset.jobId as string);
+    ReactDOM.render(
+      <RootContainer>
+        <JobReviewHrDataFetcher jobId={jobId} />
+      </RootContainer>,
+      container,
```
Comment (yonikid15) [inline resources/assets/js/components/HRPortal/JobReviewHrPage.tsx]: Is there a reason you needed to brake this into 2 components (JobReviewHrPage and JobReviewDataFetcher)?

File: resources/assets/js/components/HRPortal/JobReviewHrPage.tsx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+          tasks={keyTasks}
+          criteria={criteria}
+          skills={skills}
+          departments={departments}
+          hideBuilderLinks={true}
+        />
+      ) : (
+        <div data-c-alignment="base(centre)">
+          <i className="fa fa-spinner fa-spin" />
+        </div>
```
Comment (yonikid15) [inline resources/assets/js/components/HRPortal/JobReviewHrPage.tsx]: This icon should meet accessible standards. I can do this real quick though. I reference this https://fontawesome.com/how-to-use/on-the-web/other-topics/accessibility

Comment (tristan-orourke) [reply resources/assets/js/components/HRPortal/JobReviewHrPage.tsx]: Oh, thanks! Maybe we should build a reusable loading spinner component. Would make it easier to customize consistently too.

File: resources/assets/js/components/HRPortal/JobReviewHrPage.tsx (hunk 1 above)
Comment (tristan-orourke) [reply resources/assets/js/components/HRPortal/JobReviewHrPage.tsx]: Separation of concerns. The data fetching logic can be a bit tricky, and there's a bunch of it that needs to happen all at once here. It seems neater to me to separate that logic from the rest of the component.

File: resources/assets/js/components/HRPortal/JobReviewHrPage.tsx (hunk 2 above)
Comment (yonikid15) [reply resources/assets/js/components/HRPortal/JobReviewHrPage.tsx]: Ya that's a good idea. Should I make a new issue or did you want to build one on this branch?

**Pull request merged**